src/internal/connector/graph/middleware.go

```diff
@@ -209,13 +209,23 @@ func (mw RetryMiddleware) Intercept(
 	ctx := req.Context()
 
 	resp, err := pipeline.Next(req, middlewareIndex)
-	if err != nil && !IsErrTimeout(err) && !IsErrConnectionReset(err) {
+
+	// Alternate approach
+	retriable := IsErrTimeout(err) || IsErrConnectionReset(err) ||
+		(resp != nil && (resp.StatusCode/100 == 4 || resp.StatusCode/100 == 5))
+
+	if !retriable {
 		return resp, stackReq(ctx, req, resp, err)
```

Comment:
@ryanfkeepers Can I get your thoughts on this? :) I'm seeing `200, no retries` test fail with this alternate approach. 
Even though `stackReq` returns nil if err is nil, for some reason, the error check in middleware_test.go, line 153 fails. Happy to provide more context